OneCommunityGlobal/HighestGoodNetworkApp
# Luis add warnings component to dashboard

# Description
<img width="863" alt="Screenshot 2024-01-23 at 8 49 55 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/7dae6db3-a541-4c6f-82d7-527493393a25">

The main functions I created was: 
A way to track warnings issued by an Admin or Owner account. 
8 Icons are displayed and can be selected assigning a warning
After 2 warnings assigend a modal will appear to provide options.
Issuing a warning will change the color to yellow (Future PR will send an email currently only an alert is issued)
Issuing a blue square will change the color to red (Future PR will create and assign a blue square)
Selecting a colored icon will delete it, a modal will appear to confirm the deletion. 


## Related PRS (if any):
This frontend PR is related to the https://github.com/OneCommunityGlobal/HGNRest/pull/718 backend PR.
…

## Main changes explained:
Created a warnings component inside of components
All notable files related to the component are inside of warnings. 
`Warnings.jsx` is the main file and maps over the user's warnings received from the backend.
The warnings are also filtered by blue yellow red then by date. 


## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Login as an admin account.
4. Once the dashboard loads, you'll see a toggle button underneath every user
5. click the toggle button to view the users' warnings
6. Click an icon to assign a warning to the user' corresponding to the description. 
7. Once they receive more than 2 a modal will appear. 
8. 4 options are given issue warning, log warning only, cancel and issue blue square
9. log warning only will color the icon blue like normal
10. issue warning will color the icon yellow an alert will appear saying an email was sent (Future PR)
11. Issuing a blue square will color red and an alert will appear saying blue square was sent (Future PR)
12. Selecting a colored allows you to delete it.
13. You may toggle the button to remove the warnings component and will appear one you click it again. 

## Screenshots or videos of changes:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/ff425741-6b7f-4472-a5cb-cec7f3e2d56e


## Note:
This is the basis for the warnings component. 
I'll create other PR to add more warning descriptions, send emails, and issue blue square in the future. I just wanted to lay out a basis first then add to it. 
the following warnings appear and are unrelated to the PR 
<img width="349" alt="Screenshot 2024-01-23 at 8 20 35 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/274faedf-b48b-4f06-831d-fdeb6fdf6ce0">
